src/Sarif/Writers/SarifLogger.cs

```diff
@@ -447,6 +449,8 @@ private void CaptureArtifact(ArtifactLocation fileLocation)
                 fileLocation.Uri = null;
                 fileLocation.UriBaseId = null;
             }
+
+            _insertOptionalDataVisitor?.VisitArtifactLocation(fileLocation);
```

Comment:
>_insertOptionalDataVisitor [](start = 12, length = 26)

Is this strictly necessary? you call VisitRun(run) after analysis completes, won't this have the effect of ultimately visiting every artifact location?

Just curious. :)